test/fixtures/terminal/index.js

```diff
@@ -1,32 +1,19 @@
 /* global Pear */
-const { config,versions } = Pear
-const [grn, rst, dim] = ['\x1b[32m', '\x1b[0m', '\x1b[2m']
-const v = ({ key, length, fork }) => `v${fork}.${length}.${(key += '').length <= 12 ? key : key.slice(0, 12) + '…'}`
-const { app, platform } = await versions()
+const { teardown } = Pear
+import nodeInspector from 'bare-inspector'
+import { Inspector } from 'pear-inspect'
+import Pipe from 'bare-pipe'
 
-const debug = { ready: false, update: false }
+const stdout = new Pipe(1)
+stdout.unref()
 
-const argv = global.Bare?.argv || global.process.argv
-const debarg = argv.find(arg => arg.startsWith('--debug='))
+const inspector = new Inspector({ inspector: nodeInspector})
+const inspectorKey = await inspector.enable()
 
-if (debarg) {
-  const opts = debarg.split('=')[1].split(',')
-  debug.ready = opts.includes('ready')
-  debug.update = opts.includes('update')
+global.endInspection = async function () {
+  await inspector.disable()
 }
 
-const out = `${grn}           ▅
-           ▀
-        ▂▂▄▟▙▃
-       ▄▄▄▄▆▆▆▆         ${config.name}
-      ▄▄▄▄▄▆▆▆▆▆        ${dim}${v(app)}${rst}${grn}
-      ▄▄▄▄▄▆▆▆▆▆
-     ▄▄▄▄▄▄▆▆▆▆▆▆       ${rst}${grn}pear
-    ▃▄▄▄▄▄▄▆▆▆▆▆▆▄      ${dim}${v(platform)}${rst}${grn}
-   ▄▄▄▄▄▄▄▄▆▆▆▆▆▆▆▆
-   ▄▄▄▄▄▄▄▄▆▆▆▆▆▆▆▆     ${rst}${grn}Welcome to the IoP
-     ▄▄▄▄▄▄▆▆▆▆▆▆
-       ▄▄▄▄▆▆▆▆
-`
-console.log('\n\x1b[s\x1b[J' + out + '\x1b[0m')
-if (debug.ready) console.log('[DEBUG] READY')
+stdout.write(inspectorKey.toString('hex'))
```

Comment:
why not console.log ?

Reply:
when running the app directly from the terminal, console.log logs as expected.However, when the app is run as a child process (not directly on a terminal) the logs are being buffered and not flushed to the parent until the child process exits. Bare Pipe, on the other hand, outputs directly.